### PR TITLE
Fix all broken links and add link checker

### PR DESCRIPTION
**Pull request by bravegrape**
### Context

As a result of the change in docs structure (https://github.com/alphagov/tdt-documentation/pull/97), a number of links broke. 

### Changes proposed in this pull request

Manually fix broken links and add a link checker such that the build will fail if there are any broken links. This should make sure future changes to the docs will not lead to more broken links.

The PR also copies the docs content from the tech docs gem: https://github.com/alphagov/tech-docs-gem/tree/master/docs. This content is currently copied over and updated on every build through the `gem_docs` helper method. The fragmented setup makes it hard to make changes that involve that content. The plan is to eventually move everything in the `alphagov/tdt-documentation` repo to the tech docs gem repo. 

~There will be a PR signposting the content move in `alphagov/tech-docs-gem/docs`~ This is the PR to add signposting for the content move: https://github.com/alphagov/tech-docs-gem/pull/151

### Guidance to review

Run build locally. `html-proofer` shouldn't find any broken links.

### CODE REVIEW

**Comment by 36degrees** [comment]
A very minor (and existing) thing that'd be good to fix at some point but shouldn't block this – there's a mix of wrapped and non-wrapped lines within and across all the files. It'd be good to try and make it consistent at some point.

File: source/amend_project/configuration/index.html.md.erb [hunk 1]
```diff
@@ -7,4 +7,229 @@ review_in: 1 day
 
 # Configuration options
 
-<%= gem_docs 'configuration.md' %>
```
Comment (nickcolley) [inline source/amend_project/configuration/index.html.md.erb]: Does this mean we need to do a pull request on the gem repo to remove content?

Comment (36degrees) [reply source/amend_project/configuration/index.html.md.erb]: > This is the PR to add signposting for the content move: alphagov/tech-docs-gem#151

Comment (bravegrape) [reply source/amend_project/configuration/index.html.md.erb]: Yup - I've raised this PR to cover the content move: https://github.com/alphagov/tech-docs-gem/pull/151

**Comment by bravegrape** [comment]
> A very minor (and existing) thing that'd be good to fix at some point but shouldn't block this – there's a mix of wrapped and non-wrapped lines within and across all the files. It'd be good to try and make it consistent at some point.

@36degrees I've raised an issue to keep track of this: https://github.com/alphagov/tdt-documentation/issues/109

**Pull request merged**